packages/e2e/suites/full/freeze.spec.ts

```diff
@@ -0,0 +1,536 @@
+  page,
+  switcher,
+}) => {
+  await switcher.council(1, page);
+
+  await goToFreezeIndex(page);
+
+  await page.getByTestId("unfreeze-create-btn").click();
+
+  await expect(page.getByText("Create Unfreeze Proposal")).toBeVisible();
+  await expect(page.getByText("Valid Until")).toBeVisible();
+  await expect(page.getByRole("button", { name: "Create" })).toBeEnabled();
+
+  await page.getByRole("button", { name: "Create" }).click();
+
+  await page.waitForURL("**/app/freeze/*");
+
+  const now = new Date();
+  const oneWeekFromNow = new Date(Date.now() + 1000 * 3600 * 24 * 7);
+  const validUntilText = await page.getByTestId("valid-until-timestamp").textContent();
+  const proposedOnText = await page.getByTestId("proposed-on-timestamp").textContent();
+  compareExtractedTextWithDate(validUntilText, oneWeekFromNow);
+  compareExtractedTextWithDate(proposedOnText, now);
+});
+
+test("TC333: Attempt to approve unfreeze by guardian -> Sign button disabled", async ({
+  page,
+  switcher,
+}) => {
+  await switcher.guardian(1, page);
+  await assertCannotSignProposal(page, "unfreeze");
+});
+
+test("TC334: Attempt to approve unfreeze by zk foundation -> Sign button disabled", async ({
+  page,
+  switcher,
+}) => {
+  await switcher.zkFoundation(page);
+  await assertCannotSignProposal(page, "unfreeze");
+});
+
+test("TC335: Attempt to approve unfreeze by visitor -> Sign button disabled", async ({
+  page,
+  switcher,
+}) => {
+  await switcher.visitor(page);
+  await assertCannotSignProposal(page, "unfreeze");
+});
+
+test("TC332: Approve change unfreeze by security council -> Updates state", async ({
+  page,
+  switcher,
+  wallet,
+}) => {
+  await switcher.council(1, page);
+  await createFreeze(page, "unfreeze");
+  await goToFreezeDetailsPage(page, "unfreeze");
+
+  await withVoteIncrease(page, async () => {
+    await approveFreeze(page, wallet);
+  });
+
+  await expect(approveButton(page)).toBeDisabled();
+});
+
+test("TC336, TC337: Freeze -> Gather unfreeze signatures -> Exec unfreeze tx -> tx is correctly created and displayed", async ({
+  page,
+  switcher,
+  wallet,
+}) => {
+  await createFreeze(page, "soft");
+  await applyApprovals(page, "soft", switcher, wallet, [1, 2, 3]);
+  await page.getByRole("button", { name: "Execute freeze" }).click();
+  await wallet.confirmTransaction();
+
+  await createFreeze(page, "unfreeze");
+  await applyApprovals(page, "unfreeze", switcher, wallet, [1, 2, 3, 4, 5, 6, 7, 8, 9]);
+  await broadcastAndCheckFreeze(page, wallet);
+});
+
+// change threshold
+
+async function cannotSignChangeThreshold(page: Page) {
+  await createChangeThreshold(page);
+
+  await goToFreezeDetailsPage(page, "change-threshold");
+  await expect(page.getByText("No role actions")).toBeVisible();
+  const approveButtons = await page.getByRole("button", { name: "Approve" }).all();
+  expect(approveButtons).toHaveLength(0);
+}
```

Comment:
Can't we reuse the `assertCannotSignProposal` function here?

Reply:
Well... Maybe. The issue is that those are slightly different in an awful way. I wanted to have linear code for this tests. Avoid conditionals as much as possible.

If I accept adding conditionals I can remove the duplicated functions for change threshold. Let me try it to see how it looks.